x/gov/codec.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package gov;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "github.com/iov-one/weave/coin/codec.proto";
+import "google/protobuf/timestamp.proto";
+
+// Electorate defines who may vote in an election. This same group can be used in many elections
+// and is stored for re-use
+message Electorate {
+    // Participants defines a list of all signatures that are allowed to participate in a vote
+    repeated Participant participants = 1 [(gogoproto.nullable) = false];
+}
+
+// Participant clubs together a signature with a weight. The greater the weight
+// the greater the power of a participant.
+message Participant {
+    bytes signature = 1 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+    uint32 power = 2 [(gogoproto.casttype) = "Weight"];
+}
+
+// Election Rules defines how an election is run. A proposal must be voted upon via a pre-defined ruleset.
+message ElectionRules {
+    // Human readable title.
+    string title = 1;
+    // Reference to the electorate to define the group of possible voters.
+    bytes electorate_id = 2;
+    // Duration how long the voting period will take place.
+    uint32 voting_period_hours = 3;
+    // Required Vote this is percentage of all eligible voters, not the ones who voted.
+    // (integer in steps of 0.01%, so 10000 is 100%. should disallow anything below 5001)
+    uint32 required_votes = 4;
+}
+
+// A text form proposal for an on-chain governance process.
+message TextProposal {
+    // Human readable title.
+    string title = 1;
+    // Description of the proposal in text form.
+    string description = 2;
+    // Reference to the election rules
+    bytes election_rules_id = 3;
+    // Time of the block where the voting period starts. Header time of the votes must be greater than this
+    // start time.
+    google.protobuf.Timestamp voting_start_time = 4 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
```

Comment:
We never need nanosecond precision.
And can probably get away with hour precision.
But I think measuring timestamps in seconds (unix time) is quite standardized

Reading another comment about secondary indexes... I think we will often want to query for voting_end_time > X. The index must return a simple field (eg. seconds), stored in a fixed length, so that times are lexographically ordered (byte gte comparisons on raw indexes will be valid for the numeric interpretation as well)